mautic/mautic
# Add DNS based and plugin integration based validation for emails

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| New feature? | y
| Related user documentation PR URL | todo
| Related developer documentation PR URL | todo

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR adds email validation based on MX records. It currently does not do this automatically as I'm not fully sure what the implications will be. However, it adds a campaign condition that can be used to tag, segment, mark as DNC, etc. 

It also cleans up emails with `..`, or spaces. And it adds back the lost email validation on contact import. 

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Create a campaign with the new Has valid email condition. Tag if they do and tag something else if they don't. 
2. Be sure a couple contacts have non-existent domains like `sssssss.com`
3. Run the campaign - contacts should be tagged appropriately. Note that this does not out of the box determine if the user exists, only if the domain has email MX records
4. Import a CSV of emails both good and bad
5. Bad ones should be noted in the output
6. Run the new tests
